dnf/conf/read.py

```diff
@@ -54,17 +54,37 @@ def __iter__(self):
     def _build_repo(self, parser, id_, repofn):
         """Build a repository using the parsed data."""
 
-        repo = dnf.repo.Repo(id_, self.conf)
+        substituted_id = libdnf.conf.ConfigParser.substitute(id_, self.conf.substitutions)
+
+        # Check the repo.id against the valid chars
+        invalid = dnf.repo.repo_id_invalid(substituted_id)
+        if invalid is not None:
+            if substituted_id != id_:
+                msg = _("Bad id for repo: %s (%s), byte = %s %d") % \
```

Comment:
We are switching from % string formatting to .format(). Can you please replace it with
```
_("Bad id for repo: {} ({}), byte = {} {}").format(substituted_id, id_, substituted_id[invalid], invalid)
```
Otherwise the change LGTM.